sdk/core/core-rest-pipeline/src/util/stream.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Readable } from "stream";
+import { BlobLike, FileLike } from "../interfaces";
+
+export function isNodeReadableStream(x: unknown): x is NodeJS.ReadableStream {
+  return Boolean(x && typeof (x as NodeJS.ReadableStream)["pipe"] === "function");
+}
+
+export function isWebReadableStream(x: unknown): x is ReadableStream {
+  return Boolean(
+    x &&
+      typeof (x as ReadableStream).getReader === "function" &&
+      typeof (x as ReadableStream).tee === "function"
+  );
+}
+
+function nodeStreamFromWebStream(webStream: ReadableStream): NodeJS.ReadableStream {
+  //grumble
+  return Readable.from(
```

Comment:
I'm surprised the webStream itself isn't iterable and able to be handled by `Readable.from()` directly.

Not sure if `stream.pipeline` could help us here

Reply:
I mentioned this offline but in case anyone else is curious: Chromium web streams aren't async iterable yet, while Node ones are. An interesting consequence of this is that there currently two diverging type definitions of `ReadableStream`: the one in the global namespace, without `@@asyncIterator`, and the one in `node:streams/web`, which is identical except for having async iterator support.

- Relevant DT discussion: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/65542
- Chromium bug: https://bugs.chromium.org/p/chromium/issues/detail?id=929585